transformer_engine/common/comm_gemm_overlap/comm_gemm_overlap.cpp

```diff
@@ -557,7 +557,7 @@ CommOverlapP2PBase::CommOverlapP2PBase(const std::vector<size_t> &buffer_shape,
 
   // Create workspace tensor with userbuffer
   NVTE_CHECK(buffer_shape.size() == 2, "Userbuffer shape must be 2-dimensional!");
-  size_t buffer_bytes = buffer_shape[0] * buffer_shape[1] * typeToSize(buffer_dtype);
+  size_t buffer_bytes = static_cast<size_t>(buffer_shape[0] * buffer_shape[1] * typeToSize(buffer_dtype));
```

Comment:
I would say that expressions like this should be a separate function, checking whether the size is divisible by 2 when type is fp4.